packages/compass-e2e-tests/tests/search-indexes.test.ts

```diff
@@ -0,0 +1,188 @@
+  });
+
+  afterEach(async function () {
+    // Drop the collection
+    {
+      try {
+        await dbInstance.dropCollection(collectionName);
+      } catch (e) {
+        console.log(`Failed to drop collection: ${DB_NAME}.${collectionName}`);
+      }
+    }
+    void mongoClient.close();
+    await disconnect(browser);
+    await afterTest(compass, this.currentTest);
+  });
+
+  for (const { name, connectionString } of connectionsWithNoSearchSupport) {
+    context(`does not support search indexes in ${name}`, function () {
+      before(function () {
+        if (!connectionString) {
+          return this.skip();
+        }
+        currentConnectionString = connectionString;
+      });
+
+      it('allows users to create a regular index', async function () {
+        const indexName = await browser.createIndex({
+          fieldName: 'e2e_tests_index',
+          indexType: 'text',
+        });
+        await browser.dropIndex(indexName);
+      });
+
+      it('renders search indexes tab disabled', async function () {
+        const searchTab = await browser.$(
+          Selectors.indexesSegmentedTab('search-indexes')
+        );
+        const isTabClickable = await searchTab.isClickable();
+        expect(isTabClickable).to.be.false;
+      });
+    });
+  }
+
+  for (const { name, connectionString } of connectionsWithSearchSupport) {
+    context(`supports search indexes in ${name}`, function () {
+      before(function () {
+        if (!connectionString) {
+          return this.skip();
+        }
+        currentConnectionString = connectionString;
+      });
+
+      it('allows users to create a regular indexes', async function () {
+        await browser.clickVisible(
+          Selectors.indexesSegmentedTab('regular-indexes')
+        );
+        const indexName = await browser.createIndex({
+          fieldName: 'e2e_tests_index',
+          indexType: 'text',
+        });
+        await browser.dropIndex(indexName);
+      });
+
+      it('allows users to create, view and drop search indexes', async function () {
+        const indexName = `e2e_search_index_${getRandomNumber()}`;
+        await browser.clickVisible(
+          Selectors.indexesSegmentedTab('search-indexes')
+        );
+        await browser.createSearchIndex(indexName, INDEX_DEFINITION);
+        await browser.waitForAnimations(Selectors.SearchIndexList);
+
+        const rowSelector = Selectors.searchIndexRow(indexName);
+
+        // View it
+        await browser.$(rowSelector).waitForDisplayed();
+        await browser.waitUntil(async function () {
+          const text = await browser.$(rowSelector).getText();
+          return text.indexOf('PENDING') > -1;
+        });
+
+        // Drop it
+        await browser.dropSearchIndex(indexName);
+        await browser.waitUntil(async function () {
+          const text = await browser.$(rowSelector).getText();
+          return text.indexOf('DELETING') > -1;
```

Comment:
I think this will be flaky because waitUntil polls and deleting could be there only for a split second. Plus we'll poll to update the list as well so the row could be gone before we ever see deleting. If they ever speed things up it could be gone by the time we first refresh the list too.

So I think safest here would be to either only do this once we poll and then wait for the row to be gone entirely OR you'd have to check for either the status to become DELETING OR the row to disappear. Maybe a Promise.race()?

Reply:
In more general terms: If you're going to wait for something in an e2e test you have to wait for a stable state that won't change until the user intervenes. Waiting for an intermediate state that will change by itself will cause flakyness down the line. If you're lucky this works until we add polling and then hopefully it breaks reliably. Otherwise it becomes the kind of thing that breaks rarely and is hard to debug.